### PR TITLE
Implement mapping for access (H2 specific). Changes approach to addin…

### LINKED ISSUE #2008: Implement mapping for access (H2 specific)
See spec/services/cocina/mapping/descriptive/h2/access_h2_spec.rb.

### PR DESCRIPTION
**Pull request by justinlittman**
…g digital repository.

closes #2008

## Why was this change made?
More mappings. Add digital repository whenever PURL present.

## How was this change tested?
Unit, sdr-deploy

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

**Comment by justinlittman** [comment]
Before:
```
Status (n=100000):
  Success:   98293 (98.293%)
  Different: 1158 (1.158%)
  To Cocina error:     18 (0.018%)
  To Fedora error:     0 (0.0%)
  Missing:     531 (0.531%)
```

After:
```
Status (n=100000):
  Success:   98293 (98.293%)
  Different: 1158 (1.158%)
  To Cocina error:     18 (0.018%)
  To Fedora error:     0 (0.0%)
  Missing:     531 (0.531%)
```

**Review by arcadiafalcone — APPROVED** [review]

**Comment by justinlittman** [comment]
Needs infra dev review.

File: app/services/cocina/from_fedora/descriptive/access.rb [hunk 1]
```diff
@@ -139,7 +132,7 @@ def descriptive_value_for(nodes)
                 uri: Authority.normalize_uri(node[:authorityURI])
               }.compact
               attrs[:source] = source unless source.empty?
-              attrs[:type] = node[:type]
+              attrs[:type] = type || node[:type]
```
Comment (ndushay) [inline app/services/cocina/from_fedora/descriptive/access.rb:135]: i believe got bitten by an expression like this -- won't it return a boolean value?   Do you have tests specifically for each side and both sides of this expression being true?

**Review by ndushay — CHANGES_REQUESTED** [review]
I have one question / concern.

File: app/services/cocina/from_fedora/descriptive/access.rb (hunk 1 above)
Comment (justinlittman) [reply app/services/cocina/from_fedora/descriptive/access.rb:135]:
```
irb(main):001:0> x = 'foo'
=> "foo"
irb(main):002:0> y = nil
=> nil
irb(main):003:0> x || y
=> "foo"
irb(main):004:0> y || x
=> "foo"
```

And if I take out either side of `type || node[:type]`, then tests fail.

Comment (ndushay) [reply app/services/cocina/from_fedora/descriptive/access.rb:135]: okay.  who knows what the heck I'm remembering.

**Review by ndushay — APPROVED** [review]

**Pull request merged**